tests/plugins/dialog/plugin.js

```diff
@@ -1,347 +1,407 @@
+								assert.isFalse( dialog.getButton( 'ok' ).isEnabled(), 'OK button is disabled' );
+								assert.isObject( dialog.parts.spinner, 'Dialog has a spinner element' );
 
-							++stateEventFired;
-						} catch ( e ) {
-							evt.removeListener();
-							throw e;
-						}
-					} );
+								++stateEventFired;
+							} catch ( e ) {
+								evt.removeListener();
+								throw e;
+							}
+						} );
 
-					// Change dialog's state and assert related properties.
-					dialog.setState( CKEDITOR.DIALOG_STATE_BUSY );
+						// Change dialog's state and assert related properties.
+						dialog.setState( CKEDITOR.DIALOG_STATE_BUSY );
 
-					// Remove the listener because the dialog will be reopened and those assertions would be invalid.
-					stateListener.removeListener();
+						// Remove the listener because the dialog will be reopened and those assertions would be invalid.
+						stateListener.removeListener();
 
-					assert.areSame( 1, stateEventFired, 'State event has been fired' );
+						assert.areSame( 1, stateEventFired, 'State event has been fired' );
 
-					dialog.hide();
+						dialog.hide();
 
-					// Call the dialog again to tell what happens to the state and the UI once reopened.
-					editor.execCommand( 'testDialog5' );
+						// Call the dialog again to tell what happens to the state and the UI once reopened.
+						editor.execCommand( 'testDialog5' );
 
-					editor.once( 'dialogShow', function( evt ) {
-						var dialog = evt.data;
+						editor.once( 'dialogShow', function( evt ) {
+							var dialog = evt.data;
 
-						resume( function() {
-							try {
-								assert.areSame( CKEDITOR.DIALOG_STATE_IDLE, dialog.state, 'Default dialog state after re–open' );
-								assert.isTrue( dialog.getButton( 'ok' ).isEnabled(), 'OK button is enabled after re–open' );
-								assert.isObject( dialog.parts.spinner, 'Dialog has been given a spinner before' );
-							} catch ( e ) {
-								throw e;
-							} finally {
-								dialog.hide();
-							}
+							resume( function() {
+								try {
+									assert.areSame( CKEDITOR.DIALOG_STATE_IDLE, dialog.state, 'Default dialog state after re–open' );
+									assert.isTrue( dialog.getButton( 'ok' ).isEnabled(), 'OK button is enabled after re–open' );
+									assert.isObject( dialog.parts.spinner, 'Dialog has been given a spinner before' );
+								} catch ( e ) {
+									throw e;
+								} finally {
+									dialog.hide();
+								}
+							} );
 						} );
-					} );
 
-					wait();
-				} catch ( e ) {
-					throw e;
-				} finally {
-					dialog.hide();
-				}
+						wait();
+					} catch ( e ) {
+						throw e;
+					} finally {
+						dialog.hide();
+					}
+				} );
 			} );
-		} );
 
-		editor.execCommand( 'testDialog5' );
+			editor.execCommand( 'testDialog5' );
+
+			wait();
+		},
 
-		wait();
-	}
-} );
+		// #830
+		'test dialog opens non first tab as default': function() {
```

Comment:
Maybe "test dialog opens tab defined in tabId as default"?

Reply:
good point it has more sense 👍